go/vt/vtgate/engine/primitive.go

```diff
@@ -260,6 +267,12 @@ type (
 
 	// txNeeded is a default implementation for Primitives that need transaction handling
 	txNeeded struct{}
+
+	PrimitiveID int
+
+	identifiablePrimitive struct {
+		id PrimitiveID
+	}
 )
```

Comment:
I think we can get away without having this to be added to all the primitives and can use primitive address pointer to tract them.